### PR TITLE
Allow --from-unlocked flag to specify when only uncollected staking rewards and unlocked substakes are used for creating/increasing a stake.

### LINKED ISSUE #2273: Clarify staking liquid vs. escrow-unlocked NU.
Current interactive prompt when staking via CLI:

```
Transfer tokens from the staker address? Otherwise, unlocked tokens in the escrow will be used [True]
```

### PR DESCRIPTION
**Pull request by derekpierre**
Fixes #2273 .

### CODE REVIEW

File: nucypher/cli/commands/stake.py [hunk 1]
```diff
@@ -451,8 +454,8 @@ def unbond_worker(general_config, transacting_staker_options, config_file, force
 @option_value
 @option_lock_periods
 @group_general_config
-@option_only_lock
-def create(general_config, transacting_staker_options, config_file, force, value, lock_periods, only_lock):
+@option_use_uncollected_rewards
```
Comment (KPrasch) [inline nucypher/cli/commands/stake.py:457]: :+1:  How about "unclaimed" throughout; `--from-unclaimed-rewards`

File: tests/integration/cli/test_stake_cli_functionality.py [hunk 2]
```diff
@@ -977,10 +978,11 @@ def test_create_lock_interactive(click_runner,
 
     command = ('create',
                '--provider', MOCK_PROVIDER_URI,
-               '--network', TEMPORARY_DOMAIN)
+               '--network', TEMPORARY_DOMAIN,
+               '--use-rewards')
 
     user_input = '\n'.join((str(selected_index),
-                            NO,
+                            YES,
```
Comment (KPrasch) [inline tests/integration/cli/test_stake_cli_functionality.py:985]: Haha - this is a funny--to-read diff line :joy:

File: nucypher/cli/commands/stake.py (hunk 1 above)
Comment (derekpierre) [reply nucypher/cli/commands/stake.py:457]: 🤔 Interestingly, I had `unclaimed` before, but I realized that we only use "claim" for worklock in our code - "collect" was used in the context of rewards.

I'm fine to change it if desired, but was trying to keep the parlance the same.

Comment (derekpierre) [reply nucypher/cli/commands/stake.py:457]: Using "from" instead of "use" is good

Comment (derekpierre) [reply nucypher/cli/commands/stake.py]: ✔️

File: nucypher/cli/commands/stake.py [hunk 3]
```diff
@@ -102,13 +101,17 @@
...
+from nucypher.cli.utils import setup_emitter
 from nucypher.config.characters import StakeHolderConfiguration
 
 option_value = click.option('--value', help="Token value of stake", type=click.INT)
 option_lock_periods = click.option('--lock-periods', help="Duration of stake in periods.", type=click.INT)
 option_worker_address = click.option('--worker-address', help="Address to bond as an Ursula-Worker", type=EIP55_CHECKSUM_ADDRESS)
 option_index = click.option('--index', help="The staker-specific stake index to edit", type=click.INT)
-option_only_lock = click.option('--only-lock', help="Locking tokens without deposit", is_flag=True)
+option_from_uncollected_rewards = click.option('--from-uncollected-rewards',
+                                               help="Only use uncollected staking rewards, and not tokens from staker address",
```
Comment (vzotova) [inline nucypher/cli/commands/stake.py:112]: I guess this is fine, but if to be precisely - unlocked tokens could be not only reward but also unlocked sub-stake

Comment (KPrasch) [reply nucypher/cli/commands/stake.py:112]: Yes, this is truly just "unlocked and escrowed" NU.  It's hard to describe both clearly and tersely at the same time... :thinking:

Comment (derekpierre) [reply nucypher/cli/commands/stake.py:112]: Something like `--from-alternative-source` (or if you like a slightly more descriptive name, `--from-unlocked-or-rewards`) and then use the help text to clarify? I think it'll be hard to cover the full picture in the option name.

Comment (derekpierre) [reply nucypher/cli/commands/stake.py]: Used `from-unlocked` - ✔️

**Review by vzotova — APPROVED** [review]

**Review by KPrasch — APPROVED** [review]
:cowboy_hat_face: Thank You!

**Pull request merged**